VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -161,38 +166,38 @@ public AuthResponse registerUsingFacebook(@NonNull Bundle parameters, @NonNull S
     }
 
     @NonNull
-    public ResetPasswordResponse resetPassword(@NonNull String email) throws HttpException {
-        return loginService.resetPassword(email);
-    }
-
-    @NonNull
     private void register(Bundle parameters) throws Exception {
-        try {
-            final Map<String, String> parameterMap = new HashMap<>();
-            for (String key : parameters.keySet()) {
-                parameterMap.put(key, parameters.getString(key));
-            }
-            loginService.register(parameterMap);
-        } catch (HttpResponseStatusException e) {
-            if ((e.getStatusCode() == 400 || e.getStatusCode() == 409) && !android.text.TextUtils.isEmpty(e.getBody())) {
+        final Map<String, String> parameterMap = new HashMap<>();
+        for (String key : parameters.keySet()) {
+            parameterMap.put(key, parameters.getString(key));
+        }
+        Response<ResponseBody> response = loginService.register(parameterMap).execute();
+        if (!response.isSuccessful()) {
+            final int errorCode = response.code();
+            final String errorBody = response.errorBody().string();
+            if ((errorCode == 400 || errorCode == 409) && !android.text.TextUtils.isEmpty(errorBody)) {
                 try {
-                    final FormFieldMessageBody body = gson.fromJson(e.getBody(), FormFieldMessageBody.class);
+                    final FormFieldMessageBody body = gson.fromJson(errorBody, FormFieldMessageBody.class);
                     if (body != null && body.size() > 0) {
                         throw new RegistrationException(body);
                     }
                 } catch (JsonSyntaxException ex) {
                     // Looks like the response does not contain form validation errors.
                 }
             }
-            throw e;
+            throw new HttpResponseStatusException(errorCode);
         }
     }
 
     @NonNull
     public ProfileModel getProfile() throws Exception {
```

Comment:
Same thing here. Can we make this method throws HttpResponseStatusException? I know you didn't change the signature during your work, but may be a good idea.